### PR TITLE
chore(network): stress test utils

### PR DESCRIPTION
**Pull request by eitanm-starkware**

### CODE REVIEW

**Review by eitanm-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1995)* status: all files reviewed, 5 unresolved discussions (waiting on @ShahakShama)

___
*[`crates/papyrus_network/src/bin/network_stress_test/converters.rs` line 33 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1995#-OBZD0Y42rdO1-LrjeAz:-OB_Ee7KDST8VASmiQrt:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/ca218367856f035b731cbb435f60460f9361e4ab/crates/papyrus_network/src/bin/network_stress_test/converters.rs#L33)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Extract to a constant, and calculate it by using https://doc.rust-lang.org/std/mem/fn.size_of.html and addition
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/utils.rs` line 19 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1995#-OBZDcOc1JTF8zRQinc_:-OBZZkOyC15fyBGcLR5O:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/ca218367856f035b731cbb435f60460f9361e4ab/crates/papyrus_network/src/bin/network_stress_test/utils.rs#L19)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

rename to num_messages
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/utils.rs` line 86 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1995#-OBZEDVmA3YfL_zpIyRk:-OBZatg44rKhZoXU9t7_:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/ca218367856f035b731cbb435f60460f9361e4ab/crates/papyrus_network/src/bin/network_stress_test/utils.rs#L86)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

This function both converts to u64 and serializes. IMO it shouldn't do both. If you disagree at least rename it to reflect that it does both
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/utils.rs` line 93 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1995#-OBZE4BeDe5yFt7KaTgn:-OBZaT3W-HhIDlR02AjH:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/ca218367856f035b731cbb435f60460f9361e4ab/crates/papyrus_network/src/bin/network_stress_test/utils.rs#L93)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Why not `duration_since_epoch.as_millis()
</blockquote></details>

Done.

File: crates/papyrus_network/src/bin/network_stress_test/converters.rs [hunk 1]
```diff
@@ -1,9 +1,18 @@
 use std::time::{Duration, SystemTime};
 
-struct StressTestMessage {
-    id: u32,
-    payload: Vec<u8>,
-    time: SystemTime,
+pub const METADATA_SIZE: usize = size_of::<SystemTime>() + size_of::<u32>();
```
[the remark this answers is not in the document]
Comment (ShahakShama) [reply crates/papyrus_network/src/bin/network_stress_test/converters.rs:3]: :+1:

**Review by ShahakShama — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1995)* status: all files reviewed, 3 unresolved discussions (waiting on @eitanm-starkware and @graphite-app[bot])

___
*[`crates/papyrus_network/src/bin/network_stress_test/utils.rs` line 86 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1995#-OBZEDVmA3YfL_zpIyRk:-OB_TNs2DLw_i9uSjHJG:bxegryg) ([raw file](https://github.com/starkware-libs/sequencer/blob/ca218367856f035b731cbb435f60460f9361e4ab/crates/papyrus_network/src/bin/network_stress_test/utils.rs#L86)):*
<details><summary><i>Previously, eitanm-starkware wrote…</i></summary><blockquote>

Done.
</blockquote></details>

This is not done
___
*[`crates/papyrus_network/src/bin/network_stress_test/utils.rs` line 93 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1995#-OBZE4BeDe5yFt7KaTgn:-OB_TQc1DMN5Xlq3odMo:bxegryg) ([raw file](https://github.com/starkware-libs/sequencer/blob/ca218367856f035b731cbb435f60460f9361e4ab/crates/papyrus_network/src/bin/network_stress_test/utils.rs#L93)):*
<details><summary><i>Previously, eitanm-starkware wrote…</i></summary><blockquote>

Done.
</blockquote></details>

This is not done

File: crates/papyrus_network/src/bin/network_stress_test/utils.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    time: &SystemTime,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let duration_since_epoch =
+        time.duration_since(UNIX_EPOCH).map_err(serde::ser::Error::custom)?;
+    let millis =
+        duration_since_epoch.as_millis() + u128::from(duration_since_epoch.subsec_millis());
```
[the remark this answers is not in the document]
Comment (ShahakShama) [reply crates/papyrus_network/src/bin/network_stress_test/utils.rs:130]: :+1:

**Review by ShahakShama — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r3, 3 of 3 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1995#-:-OB_ehfAC2WrSBrc_Obx:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @graphite-app[bot])

**Review by ShahakShama — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1995#-:-OB_f13F76lzw80vd_Eb:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @graphite-app[bot])

**Pull request closed without merging**